### PR TITLE
Update ReleaseProcess.md

### PR DESCRIPTION
**Pull request by matsduf**
The `make distcheck` makes sense after `make all`, not before.

### CODE REVIEW

File: docs/internal-documentation/maintenance/ReleaseProcess.md [hunk 1]
```diff
@@ -77,24 +77,25 @@ For all components, generate Makefile, META.yml and others.
 > and urge informing the author. Instead of immediately informing the author,
 > verify that META.yml was created by the command at a later stage.
 
+Create derived files (if any) and verify that a distribution tarball can be
```
Comment (mattias-p) [inline docs/internal-documentation/maintenance/ReleaseProcess.md]: I believe the usual terminology is "Build generated files".

File: docs/internal-documentation/maintenance/ReleaseProcess.md [hunk 2]
```diff
@@ -77,24 +77,25 @@ For all components, generate Makefile, META.yml and others.
...
 > verify that META.yml was created by the command at a later stage.
 
+Create derived files (if any) and verify that a distribution tarball can be 
+successfully built for each component that is to be updated in this release.
+
+    make all
+
+## 7. Produce distribution tarballs
+
+In this step, produces the *preliminary distribution tarballs*.
```
Comment (mattias-p) [inline docs/internal-documentation/maintenance/ReleaseProcess.md]: Either "In this step produce" or "This step produces".

File: docs/internal-documentation/maintenance/ReleaseProcess.md [hunk 3]
```diff
@@ -82,16 +82,17 @@ successfully built for each component that is to be updated in this release.
...
-## 7. Produce distribution tarballs
+For all components, make sure that all files are covered by MANIFEST and/or 
+MANIFEST.SKIP:
 
-In this step produce the *preliminary distribution tarballs*.
+    make distcheck
 
-For all components, make sure that all files are covered by MANIFEST and/or MANIFEST.SKIP:
+> For all components, review the MANIFEST file if in doubt or if there seems to be
+> missing or extra files.
```
Comment (mattias-p) [inline docs/internal-documentation/maintenance/ReleaseProcess.md]: Do we really need an explicit note about this?

Comment (matsduf) [reply docs/internal-documentation/maintenance/ReleaseProcess.md]: @mattias-p! Not really. I just kept the text from previous version. I'll update.

**Pull request merged**